src/components/movie-card/movie-card.jsx

```diff
@@ -1,14 +1,85 @@
 import PropTypes from "prop-types";
 import { Button, Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+
+export const MovieCard = ({ movie, onToggleFavorite }) => {
+  const token = localStorage.getItem("token");
+  const [isFavorite, setIsFavorite] = useState(false);
```

Comment:
isFavorite should be a prop that gets passed to movie-card from main-view. From main-view, you can pass a boolean prop that takes such form `user.FavoriteMovies.includes(movie._id)`

Reply:
done